src/js/providers/default.ts

```diff
@@ -1,9 +1,68 @@
 import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import { GenericObject, SourceObj, SeekRange } from 'types/generic.type';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+interface DefaultProvider {
+    state?: string;
+    supports: () => boolean;
+    play: () => void;
+    pause: () => void;
+    preload: () => void;
+    load: () => void;
+    stop: () => void;
+    volume: () => void;
+    mute: () => void;
+    seek: () => void;
+    resize: () => void;
+    remove: () => void; // removes from page
+    destroy: () => void; // frees memory
+
+    setVisibility: () => void;
+    setFullscreen: () => void;
+    getFullscreen: () => boolean;
+    supportsFullscreen: () => boolean;
+
+    // If setContainer has been set; this returns the element.
+    //  It's value is used to determine if we should remove the <video> element when setting a new provider.
+    getContainer: () => void;
```

Comment:
when these get overridden by the provider in question, they will have return values - should we type those return values as part of those extending interfaces or do it here

Reply:
This works with `setContainer(element)`. See video-actions-mixin for actual implementation.